frontend/src/scenes/cohorts/cohortUtils.ts

```diff
@@ -0,0 +1,424 @@
+        )
+    }
+    return false
+}
+
+export function isValidCohortGroup(criteria: AnyCohortGroupType): boolean {
+    return !(
+        (criteria.matchType === PROPERTY_MATCH_TYPE && !criteria.properties?.length) ||
+        (criteria.matchType === ENTITY_MATCH_TYPE && !(criteria.action_id || criteria.event_id))
+    )
+}
+
+export function createCohortFormData(cohort: CohortType, isNewCohortFilterEnabled: boolean = false): FormData {
+    const rawCohort = {
+        ...(cohort.name ? { name: cohort.name } : {}),
+        ...(cohort.description ? { description: cohort.description } : {}),
+        ...(cohort.csv ? { csv: cohort.csv } : {}),
+        ...(cohort.is_static ? { is_static: cohort.is_static } : {}),
+        ...(isNewCohortFilterEnabled
+            ? {
+                  filters: JSON.stringify(
+                      cohort.is_static
+                          ? {}
+                          : /* Overwrite value with value_property for cases where value is not a behavior enum (i.e., cohort and person filters) */
+                            applyAllNestedCriteria(cohort, (criteriaList) =>
+                                criteriaList.map(
+                                    (c) =>
+                                        ({
+                                            ...c,
+                                            ...('value_property' in c ? { value: c.value_property } : {}),
+                                            value_property: undefined,
+                                        } as AnyCohortCriteriaType)
+                                )
+                            ).filters
+                  ),
+                  groups: JSON.stringify([]),
+              }
+            : {
+                  groups: JSON.stringify(
+                      cohort.is_static
+                          ? []
+                          : cohort.groups.map((group: CohortGroupType) => ({
+                                ...group,
+                                id: undefined,
+                                matchType: undefined,
+                            }))
+                  ),
+              }),
+    }
+
+    if (!isNewCohortFilterEnabled) {
+        delete rawCohort['filters']
+    }
+
+    // Must use FormData to encode file binary in request
+    const cohortFormData = new FormData()
+    for (const [itemKey, value] of Object.entries(rawCohort)) {
+        cohortFormData.append(itemKey, value as string | Blob)
+    }
+    return cohortFormData
+}
+
+export function addLocalCohortGroupId(group: Partial<CohortGroupType>): CohortGroupType {
+    const matchType = group.action_id || group.event_id ? ENTITY_MATCH_TYPE : PROPERTY_MATCH_TYPE
+
+    return {
+        matchType,
+        id: Math.random().toString().substr(2, 5),
+        ...group,
+    }
+}
+
+export function processCohortOnSet(cohort: CohortType, isNewCohortFilterEnabled: boolean = false): CohortType {
```

Comment:
I'm guessing this is tested implicitly be the cohortlogic tests. What's the general pattern/decision behind this? It seems like a lot of utils in this file could be tested independently of the cohort logic

Reply:
There is a lot of repeating functionality used in various actions in the logic, so I refactored a bunch out of the logic and into utils. It'd be a nice to have but I don't think its too necessary to individually test these again since they're all being tested a number of times in `cohortLogic.test.ts`